src/cls/lock/cls_lock.cc

```diff
@@ -162,8 +185,78 @@ static int lock_obj(cls_method_context_t hctx,
     return -EINVAL;
   }
 
+
+  utime_t now = ceph_clock_now();
+
+  // bid related variables that we'll only assign and use if a bid is
+  // present
+  std::string lock_obj_id;
+  entity_name_t bid_source;
+  ClientBidMap* client_bid_map_ptr = nullptr;
+
+  entity_inst_t origin;
+  int r = cls_get_request_origin(hctx, &origin);
+  ceph_assert(r == 0);
```

Comment:
please log and return these errors, instead of asserting on them. same with the `cls_get_obj_info()` call below. unexpected errors here should never crash the osd